### ISSUE TITLE
Some launchers not working

### ISSUE THREAD

**Issue by roydbt** [moraroy/NonSteamLaunchers-On-Steam-Deck#5]
Origin and Humble Bundle not working.
Humble doesn't let you sign in, and Origin not even starting up.

**Comment by moraroy** [comment]
@roydbt weird ill take a look

**Comment by moraroy** [comment]
@roydbt both launchers installed fine for me using the NonSteamLaunchers App ID. with Hunble Games Collection it wont let you sign in becuase in order to sign in it opens up a browser window for you to do so

**Comment by dob3k** [comment]
Origin app is out of date. Starting for me but it's useless due to big popup to use EA app. 
EA moved to EA app and force users to use it. Real pain. Origin was better.
Humble needs access to x-scheme-handler/humble. I'm trying to figure how to bypass it

**Comment by moraroy** [comment]
@dob3k  just installed origin, i didnt get any upgrade message, i deliberately put in my code to install a certain origin and then modify its xml file to bypass the upgrade window. , as for humble bundle ill look into it, dont have an account so i couldnt test it fully.

**Comment by dob3k** [comment]
@moraroy I've done first installation with origin instead of EA app as it is pre-setted, but launched Origin and it had that big popup on my deck. I didn't really bothered with that. Just started from scratch and unchecked Origin and swapped for EA. Not a biggie for me. But it would be awesome if you could get the humble working. 
Anyways... Big thanks for all your work. Honestly hate heroic and lutris and your app is simply all I needed. Even if you not get the humble working you're still great mate.

**Comment by moraroy** [comment]
@dob3k no worries brother, ill look at the code, i think its possible on your first install the xml file didnt write so ill put a check there to make sure it gets written, also, yeah for humble im going to see what i can do about that

**Comment by betomaluje** [comment]
Hi! first of all, great work man! I'm also stuck in the humble bundle and some people (using another method) bypassed it like so (if this is of any help):

_Once all are up, use the Sign In option (on the humble app) which will authenticate Humble in a web browser. After logging in this takes you to a screen where you must "Allow" the access of the app to your account.

Pressing the "Allow" button will prompt you to select an application to handle x-scheme-handler/humble. Select to show more options and select your default web browser. This will open your web browser to a URL which is a "humble://" link - you need this text for a future step.

Note: If you didn't pre-open a browser, when you do this you'll get a blank tab. If this is what you're seeing, just click the Sign-in button in Humble App again, again show more and select your web browser, and the next tab will have your URL.

THIS STEP ASSUMES YOU CHANGED NOTHING DURING THE INSTALL. :) Next go to your terminal and run command:

/home/deck/Games/humble-games-collection/handle-humble-scheme "<LINK>"

You must replace <LINK> with the URL you captured in your web browser, including the braces. (meaning there are no braces in your final command)

When you do this you will see the Humble App authenticate and login properly.

Follow steps 10 through 12 and you're good to go._

this was taken [from this Reddit Post.](https://www.reddit.com/r/SteamDeck/comments/x7duyj/how_to_install_the_humble_games_collection_on_the/)

I hope this helps a little bit

**Comment by moraroy** [comment]
@betomaluje thanks bro for the info, im looking into it and will try to update the app asap, ive had some IRL things going on but hopefully today i can add some fixes!

**Comment by bestpersonevar** [comment]
my problem with Humble is, I click allow access, i select the default browser and it keeps opening a new window, not a new tab, and i never have am able to see what the Humble link is to copy and paste into the terminal

**Comment by moraroy** [comment]
@bestpersonevar yeah this is a really big problem with this launcher... im seriously thinking of not supporting it because its a pain just to get it to run. it requires me to make a whole other script in order to fix it... ive almost have it working on my side of things but for some reason i cant get the humble bundle to create a directory where the "fix" is supposed to happen. sorry about that brother.
